angelsrefining/control.lua

```diff
@@ -1,13 +1,16 @@
+local technology_overhaul = settings.startup["angels-enable-tech"].value
+local components_overhaul = technology_overhaul or settings.startup["angels-enable-components"].value
+
 script.on_init(function()
   if remote.interfaces["freeplay"] then -- give ore crushers
     local items_to_insert = remote.call("freeplay", "get_created_items")
     items_to_insert["burner-ore-crusher"] = (items_to_insert["burner-ore-crusher"] or 0) + 1
     remote.call("freeplay", "set_created_items", items_to_insert)
   end
 
-  if game.active_mods["bobclasses"] and remote.interfaces['freeplay'] then -- give an extra miner
+  if game.active_mods["bobclasses"] and components_overhaul and remote.interfaces['freeplay'] then -- give additional plates
     local items_to_insert = remote.call("freeplay", "get_created_items")
-    items_to_insert["burner-mining-drill"] = (items_to_insert["burner-mining-drill"] or 0) + 1 -- 2 additional plates
+    items_to_insert["iron-plate"] = (items_to_insert["iron-plate"] or 0) + 2 -- 2 additional plates
     remote.call("freeplay", "set_created_items", items_to_insert)
   end
 end)
```

Comment:
You should check the settings inside the function, and instead of making two separate sections of freeplay, you should merge them together.

```suggestion
script.on_init(function()
  if remote.interfaces["freeplay"] then
    local items_to_insert = remote.call("freeplay", "get_created_items")
    local technology_overhaul = settings.startup["angels-enable-tech"].value
    local components_overhaul = technology_overhaul or settings.startup["angels-enable-components"].value
    
    -- give ore crushers
    items_to_insert["burner-ore-crusher"] = (items_to_insert["burner-ore-crusher"] or 0) + 1

    if game.active_mods["bobclasses"] then
      -- fix character classes starting point
      if components_overhaul then
        remote.call("freeplay", "set_created_items", items_to_insert)
      end
    end
  end
end)

Reply:
erm … the `if components_overhaul` branch should be modifying the number of iron plates, adjusted the PR accordingly